docs/4.4/api-reference.md

```diff
@@ -111,62 +108,317 @@ func main() {
+  // ClusterLabels is a map of node labels (used to dynamically grant access to clusters).
+  ClusterLabels Labels // essentially map[string][]string
+}
+```
+
+A couple of these fields are worth further explanation.
+
+**Labels**: Labels are arbitrary key-value pairs that can be used to differentiate nodes, apps, or leaf clusters by key attributes. For example, `NodeLabels` might have the key `environment`, with its value set to `development`, `staging`, or `production`, according to the node's location.
+
+```go
+services.Labels{
+  "environment": utils.Strings{"development", "staging"},
+}
+```
+
+Depending on which field you put these labels in, you can allow/deny access to any nodes, apps, or leaf clusters with the given labels. These labels can be very useful in systems where you need to carefully manage access across many clusters, e.g. if you are managing clusters for several outside groups.
+
+**Rules**
+A rule consists of two parts: the resources and verbs. Here's an example of a rule describing "read only" verbs applied to the SSH `sessions` resource. Depending on if it's under `Allow` or `Deny`, it means "allow/deny users of this role the ability to read or list active SSH sessions".
+
+```go
+services.NewRule(
+  services.KindSession,
+  services.RO(), // helper function to get 'read only' verbs ("list" and "read")
+)
+```
+
+**Resources**
+
+These are all of the possible resource values, which can be found in the `services` package.
+
+```go
+KindUser              = "user"
+KindLicense           = "license"
+KindRole              = "role"
+KindAccessRequest     = "access_request"
+KindPluginData        = "plugin_data"
+KindOIDC              = "oidc"
+KindSAML              = "saml"
+KindGithub            = "github"
+KindOIDCRequest       = "oidc_request"
+KindSAMLRequest       = "saml_request"
+KindGithubRequest     = "github_request"
+KindWebSession        = "web_session"
+KindAuthServer        = "auth_server"
+KindProxy             = "proxy"
+KindNode              = "node"
+KindAppServer         = "app_server"
+KindToken             = "token"
+KindCertAuthority     = "cert_authority"
+KindOIDCConnector     = "oidc"
+KindSAMLConnector     = "saml"
+KindGithubConnector   = "github"
+KindClusterConfig     = "cluster_config"
+KindSemaphore         = "semaphore"
+MetaNameClusterConfig = "cluster-config"
+KindClusterName       = "cluster_name"
+MetaNameClusterName   = "cluster-name"
+KindStaticTokens      = "static_tokens"
+MetaNameStaticTokens  = "static-tokens"
+KindTrustedCluster    = "trusted_cluster"
+KindIdentity          = "identity"
+KindKubeService       = "kube_service"
+```
+
+**Verbs**
+
+These are all of the possible resource values, which can be found in the `services` package.
+
+```go
+VerbList          = "list"
+VerbCreate        = "create"
+VerbRead          = "read"
+VerbReadNoSecrets = "readnosecrets"
+VerbUpdate        = "update"
+VerbDelete        = "delete"
+VerbRotate        = "rotate"
+```
```

Comment:
Maybe add a definition of what `readnosecrets` actually means in practice and when you'd want to use it?

(I'm sorry - I know a lot of this stuff is kinda out of scope for what you're trying to do but I'm just predicting the issues/requests that will come in afterwards)

Reply:
@webvictim Hmm I looked into it, but don't really understand well enough to write about it. I don't see comments on docs on it so I'd have to search around a lot to see how and why it's used. Would you be able to make a short description or give me some kind of starting point for research?

Reply:
@russjones @klizhentas Could you please detail what the difference is between `read` and `readnosecrets`?

Reply:
I actually found the difference while I was documenting our CA rotation, so I can add a description in the CA docs PR coming up.